packages/rum/src/apm-base.js

```diff
@@ -99,9 +99,21 @@ class ApmBase {
     return !this._disable
   }
 
+  /**
+   * When the required config keys are invalid, the agent is deactivated with
+   * logging error to the console
+   */
   config(config) {
-    var configService = this.serviceFactory.getService('ConfigService')
-    configService.setConfig(config)
+    const configService = this.serviceFactory.getService('ConfigService')
+    const errors = configService.validate(config)
+    if (errors.length === 0) {
+      configService.setConfig(config)
+    } else {
+      const loggingService = this.serviceFactory.getService('LoggingService')
+      const message = 'RUM Agent configuration is invalid: ' + errors.join(', ')
+      configService.setConfig({ active: false })
+      loggingService.error(message)
```

Comment:
Maybe use `warn` instead of `error` to avoid breaking users tests in dev environment.

Reply:
I thought about a bit, But it's actually an error. Agent wont work which is a serious message and we need the users to take action. `warnings` might be ignored silently.